### PR TITLE
Use unique identifier for Kubernetes resource names (REVISED)

### LINKED ISSUE #18: Use unique identifier for Kubernetes resource names
**Issue**
When creating a Deployment, Service, Route etc, hardcoded strings are being used for resource names and labels. This will make it difficult to create multiple deployments.

**Expected**
Generate unique identifier names on every deployment, and use them across all resources.

Running `kodo deploy ...` twice with the same image should get us two unique sets of Deployments/Services/Routes etc.

### PR DESCRIPTION
**Pull request by jaideepr97**
Fixes https://github.com/cli-playground/kodo/issues/18

### Summary (REVISED)

- Updated unique identifier to be a single string used for all k8 resources
- Split creation of deployment/service and route YAMLs into individual functions for easy testing
- included unit test cases for all functions ( except those that require client creation and cluster interactions)

### UPADTED REVISION

- unit tests for deploy/service/route generation using fake clients done 

- includes changes requested in https://github.com/cli-playground/kodo/issues/13 (previously addressed in https://github.com/cli-playground/kodo/pull/17) 

### Testing 

Co-author : Shraddha 
```
./bin/kodo deploy --token=TOKEN --server=SERVER_URL --image=openshift/hello-openshift:latest --replicas=10 --port=8080 --namespace=namespace
```

### CODE REVIEW

File: pkg/kodo/cmd/deploy.go [hunk 1]
```diff
@@ -16,7 +23,25 @@ type DeploymentVariables struct { //New struct for deployment creation variables
 	Port     int32
 }
 
-func Deploy(deployVar *DeploymentVariables, envVar *EnvironmentVariables) error {
+type DeploymentIdentifiers struct { //New struct to hold unique identifiers for deployment/service/route
+	DeploymentName string
```
Comment (sbose78) [inline pkg/kodo/cmd/deploy.go]: Do we need separate names/variables for every resource?

Comment (jaideepr97) [reply pkg/kodo/cmd/deploy.go]: I thought it would be better practice and it might be confusing otherwise
should I make them the same and create a new PR?

Comment (sbose78) [reply pkg/kodo/cmd/deploy.go]: Generate one unique identifier and use it for all objects.

What do you think?

Comment (jaideepr97) [reply pkg/kodo/cmd/deploy.go]: sounds good

Comment (sbose78) [reply pkg/kodo/cmd/deploy.go]: And to be clear, reviews are to be handled in the same PR. Just push to the same branch and I will show up here.

**Review by reginaelyse — APPROVED** [review]
tried it out and it works great!

**Review by reginaelyse — APPROVED** [review]

**Pull request merged**